public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -1,47 +1,56 @@
 import _ from 'lodash';
 import TableModel from 'app/core/table_model';
-import { TimeSeries, FieldType, Labels, formatLabels, QueryResultMeta } from '@grafana/data';
+import { TimeSeries, FieldType, Labels, formatLabels, QueryResultMeta, toDataFrame } from '@grafana/data';
 import { TemplateSrv } from 'app/features/templating/template_srv';
+import { PromDataSuccessResponse, PromMatrixData, TransformOptions } from './types';
+import { FetchResponse } from '@grafana/runtime';
 
 export class ResultTransformer {
   constructor(private templateSrv: TemplateSrv) {}
 
-  transform(response: any, options: any): Array<TableModel | TimeSeries> {
-    const prometheusResult = response.data.data.result;
+  transform(response: FetchResponse<PromDataSuccessResponse>, options: TransformOptions) {
+    const prometheusResult = response.data.data;
+
+    if (!prometheusResult.result) {
+      return [];
+    }
 
     if (options.format === 'table') {
-      return [
-        this.transformMetricDataToTable(
-          prometheusResult,
-          options.responseListLength,
-          options.refId,
-          options.meta,
-          options.valueWithRefId
-        ),
-      ];
-    } else if (prometheusResult && options.format === 'heatmap') {
+      const tableData = this.transformMetricDataToTable(
+        prometheusResult.result,
+        options.responseListLength,
+        options.refId,
+        options.meta,
+        options.valueWithRefId
+      );
+      return [toDataFrame(tableData)];
```

Comment:
So adding `toDataFrame` seems like the main change right? I mean I assume this works, bot not ideal as we will do `proResponse -> tableModel -> dataframe` running two different transforms and keeping the code for both. Is this to save time?

Reply:
Yeah, that's why I'm not sure if this is good enough or we don't want to do the tableModel/timeseries conversion but the dataframe. I can refactor this more so the prometheus request will be converted to dataframe.